src/node_locks.cc

```diff
@@ -0,0 +1,330 @@
+#include "node_locks.h"
+#include <memory>  // std::move
+#include "env-inl.h"
+#include "node.h"
+#include "node_internals.h"
+#include "v8.h"
+
+namespace node {
+namespace worker {
+
+using v8::Array;
+using v8::Boolean;
+using v8::Context;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::Null;
+using v8::Number;
+using v8::Object;
+using v8::Promise;
+using v8::String;
+using v8::TryCatch;
+using v8::Undefined;
+using v8::Value;
+
+LockManager* LockManager::current_ = nullptr;
+
+// static
+LockManager* LockManager::GetCurrent() {
+  if (current_ == nullptr) {
+    current_ = new LockManager();
+  }
+  return current_;
+}
+
+void LockManager::SnapshotTask::Run() {
+  HandleScope scope(env_->isolate());
+  LockManager* manager = LockManager::GetCurrent();
+
+  manager->Snapshot(env_, promise_.Get(env_->isolate()));
+}
+
+void LockRequest::Run() {
+  HandleScope scope(env_->isolate());
+  LockManager* manager = LockManager::GetCurrent();
+
+  // Let queue be origin's lock request queue.
+  auto queue = &manager->pending_requests_;
```

Comment:
Use explicit types and references. This applies to most usage of `auto` in this PR.